packages/plugin-coinbase/src/plugins/tokenContract.ts

```diff
@@ -0,0 +1,611 @@
+import { Coinbase, SmartContract } from "@coinbase/coinbase-sdk";
+import {
+    Action,
+    Plugin,
+    elizaLogger,
+    IAgentRuntime,
+    Memory,
+    HandlerCallback,
+    State,
+    composeContext,
+    generateObjectV2,
+    ModelClass,
+} from "@ai16z/eliza";
+import { initializeWallet } from "../utils";
+import { contractInvocationTemplate, tokenContractTemplate } from "../templates";
+import { ContractInvocationSchema, TokenContractSchema, isContractInvocationContent, isTokenContractContent } from "../types";
+import path from "path";
+import { fileURLToPath } from "url";
+import { createArrayCsvWriter } from "csv-writer";
+import fs from "fs";
+
+// Dynamically resolve the file path to the src/plugins directory
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const baseDir = path.resolve(__dirname, "../../plugin-coinbase/src/plugins");
+const contractsCsvFilePath = path.join(baseDir, "contracts.csv");
```

Comment:
this could be stored in a config

Reply:
Yeah will move this into a focus for all plugins in my clean / testing PR already out!